ice-rest-catalog/src/main/java/com/altinity/ice/rest/catalog/Main.java

```diff
@@ -401,6 +416,9 @@ public Integer call() throws Exception {
       logger.info("Catalog maintenance disabled (no maintenance schedule specified)");
     }
 
+    // Initialize Iceberg metrics reporter for Prometheus (singleton)
+    PrometheusMetricsReporter metricsReporter = PrometheusMetricsReporter.getInstance();
```

Comment:
some Metrics objects are passed around (like here), others are not (like MaintenanceMetrics.getInstance()). Let's settle on one approach